doc/source/qip-basics.rst

```diff
@@ -272,31 +272,168 @@ gate function returning a :class:`qutip.Qobj` and save it in the attribute ``use
+    qc.add_gate("SNOT", targets=1)
+    qc.add_gate("ISWAP", targets=[0,1])
+    qc.add_measurement("M0", targets=1, classical_store=0)
+
+    qc.draw("matplotlib", dpi=300, fig_width=4)
+
+  **Customization Examples**:
+
+    .. plot::
+      :include-source:
+
+      from qutip_qip.circuit import QubitCircuit
+      from qutip_qip.operations import Gate
+
+      # create the quantum circuit
+      qc = QubitCircuit(2, num_cbits=1)
+      qc.add_gate("CNOT", controls=0, targets=1)
+      qc.add_gate("SNOT", targets=1)
+      qc.add_gate("ISWAP", targets=[0,1])
+      qc.add_measurement("M0", targets=1, classical_store=0)
+
+      qc.draw("matplotlib", bulge=False, theme='dark', title="Plotting Quantum Circuit", dpi=300, fig_width=4)
+
+    To further explore the customization examples, refer to `link 🔗 <https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v5/quantum-circuits/matrenderer-plot.ipynb>`_.
+
+  **Customization Parameters**
+
+  .. list-table:: 
+    :header-rows: 1
+    :widths: 20 80
+
+    * - **Parameter**
+      - **Description**
+    * - ``dpi : int = 150``
+      - DPI of the figure.
+    * - ``fontsize : int = 10``
+      - Fontsize control at the circuit level, including tile 
+        and wire labels.
+    * - ``end_wire_ext : int = 2``
+      - Extension of the wire at the end of the circuit.
+    * - ``padding : float = 0.3``
+      - Padding between the circuit and the figure border.
+    * - ``gate_margin : float = 0.15``
+      - Margin space left on each side of the gate.
+    * - ``wire_sep : float = 0.5``
+      - Separation between the wires.
+    * - ``layer_sep : float = 0.5``
+      - Separation between the layers.
+    * - ``gate_pad : float = 0.05``
+      - Padding between the gate and the gate label.
+    * - ``label_pad : float = 0.1``
+      - Padding between the wire label and the wire.
+    * - ``fig_height : Optional[float] = None``
+      - Height of the figure.
+    * - ``fig_width : Optional[float] = 10``
+      - Width of the figure.
+    * - ``bulge : Union[str, bool] = True``
+      - Bulge style of the gate. Renders non-bulge gates if False.
+    * - ``align_layer : bool = False``
+      - Align the layers of the gates across different wires.
+    * - ``theme : Optional[Union[str, Dict]] = "qutip"``
+      - Color theme of the circuit. Available themes are 'qutip', 'light', 'dark' and 'modern'.
+    * - ``title : Optional[str] = None``
+      - Title of the circuit.
+    * - ``bgcolor : Optional[str] = None``
+      - Background color of the circuit.
+    * - ``color : Optional[str] = None``
+      - Controls color of accent elements (e.g., cross sign in the target node) 
+        and sets as default color of gate-label. Can be overwritten 
+        by gate-specific color.
+    * - ``wire_label : Optional[List] = None``
+      - Labels of the wires.
+    * - ``wire_color : Optional[str] = None``
+      - Color of the wires.
+
+- **Text**:
+
+  .. testcode::
```

Comment:
I found the problem @gadhvirushiraj . The indent matters here! `testcode`, `testoutput` and `**Customization Parameters**` should not have indent.